examples/Huggingface_Transformers/README.md

```diff
@@ -1,210 +1,217 @@
-
-Here is how Huggingface transfomers can be converted to Torchscript using the trace API, this has been shown in download_Transformer_models.py as well:
-
-First of all when setting the configs, the torchscript flag should be set :
+```
+mkdir Transformer_model
+mv pytorch_model_bin vocab.txt config.json Transformer_model/
+```
 
-`config = AutoConfig.from_pretrained(pretrained_model_name,torchscript=True)`
+If you'd like to download a pretrained model without fine tuning we've provided a simple helper script which will do the above for you. All you need to do is change [setup.config.json](https://github.com/pytorch/serve/blob/master/examples/Huggingface_Transformers/setup_config.json) to your liking and run
 
-When the model is loaded, we need a dummy input to pass it through the model and record the operations using the trace API:
+`python Download_transformer.models.py`
 
-```
-dummy_input = "This is a dummy input for torch jit trace"
-inputs = tokenizer.encode_plus(dummy_input,max_length = int(max_length),pad_to_max_length = True, add_special_tokens = True, return_tensors = 'pt')
-input_ids = inputs["input_ids"]
-traced_model = torch.jit.trace(model, [input_ids])
-torch.jit.save(traced_model,os.path.join(NEW_DIR, "traced_model.pt"))
-
-```
+For Torchscript support, check out [torchscript.md](torchscript.md)
 
 #### Setting the setup_config.json
 
 In the setup_config.json :
 
 *model_name* : bert-base-uncased , roberta-base or other available pre-trained models.
 
-*mode:* "sequence_classification "for sequence classification, "question_answering"for question answering and "token_classification" for token classification.
+*mode:* `sequence_classification`, `token_classification` or `question_answering`
+
+*do_lower_case* : `true` or `false` which configures the tokenizer
+
+*num_labels* : number of outputs for `sequence_classification`: 2, `token_classification`: 9  or `question_answering`: 0
 
-*do_lower_case* : True or False for use of the Tokenizer.
+*save_mode* : "torchscript" or "pretrained", this setting will be used by `Download_transformer_models.py` script as well as the handler, to download/save and load the model in Torchscript or save_pretrained mode respectively.
 
-*num_labels* : number of outputs for "sequence_classification", or "token_classification".
+*max_length* : maximum length for the  input sequences to the models, this will be used in preprocessing of the handler. Also, if you choose to use Torchscript as the serialized model  for packaging your model this length should be equal to the length that has been used during the tracing of the model using torch.jit.trace.
```

Comment:
```suggestion
*max_length* : maximum length for the  input sequences to the models, this will be used in preprocessing of the handler. Also, if you choose to use Torchscript as the serialized model for packaging your model this length should be equal to the length that has been used during the tracing of the model using torch.jit.trace.
```